### PR TITLE
Rename variables for a more intuitive documentation

### PR DESCRIPTION
**Pull request by tschmetzer**
## Description

This PR suggests a change of variable names and documentation comments in order to achieve a more intuitive documentation in python by more meaningful variable names. This PR is a follow up to #39266

The following variable names were changed:

In QgsGeometry::closestVertex
atVertex -> closestVertexIndex
beforeVertex -> previousVertexIndex
afterVertex -> nextVertexIndex

In QgsGeometry::closestSegmentWithContext
afterVertex -> nextVertexIndex
leftOf -> leftOrRightOfSegment

### CODE REVIEW

**Comment by tschmetzer** [comment]
Apparently the sip_check failed. I am confused by this failure meaning the .sip.in files are not up to date.

Nyall wrote to me: "Generation of the .sip.in files is
NOT automatic, but only happens as a result of directly calling
scripts/sipify.pl (or the sipify_all.sh script). The prepare commit
script includes a call to sipify so that the .sip.in files are
automatically updated, and there's CI tests on github actions which
ensure that the .sip.in files are up to date."

The prepare commit script was run by my commit hook and the local file ./python/core/auto_generated/geometry/qgsgeometry.sip.in has been updated and is in sync with qgsgeometry.h. From Nyall's statement above I deduct the .sip.in script is updated via CI. Also 'git diff' doesn't show any differences so I guess it is not meant to be committed. 

Do I need to commit the .sip.in file or what's the problem?

**Comment by nyalldawson** [comment]
> Do I need to commit the .sip.in file

That's correct -- you need to include the changes which have been made to the .sip.in file in your PR. The CI check doesn't automatically update them, it just checks to confirm that you've included them...

File: src/core/geometry/qgsgeometry.h [hunk 1]
```diff
@@ -630,15 +630,15 @@ class CORE_EXPORT QgsGeometry
...
+     * \param previousVertexIndex will be set to the vertex index of the previous vertex from the closest one. Will be set to -1 if
      * not present.
-     * \param afterVertex will be set to the vertex index of the next vertex after the closest one. Will be set to -1 if
+     * \param nextVertexIndex will be set to the vertex index of the next vertex after the closest one. Will be set to -1 if
      * not present.
      * \param sqrDist will be set to the square distance between the closest vertex and the specified point
      * \returns closest point in geometry. If not found (empty geometry), returns null point and sqrDist is negative.
      */
-    QgsPointXY closestVertex( const QgsPointXY &point, int &atVertex SIP_OUT, int &beforeVertex SIP_OUT, int &afterVertex SIP_OUT, double &sqrDist SIP_OUT ) const;
+    QgsPointXY closestVertex( const QgsPointXY &point, int &closestVertexIndex SIP_OUT, int &previousVertexIndex SIP_OUT, int &nextVertexIndex SIP_OUT, double &sqrDist SIP_OUT ) const;
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometry.h:641]: There's a TODO note a couple of lines above the documentation here which you can safely remove now

**Comment by nyalldawson** [comment]
@tschmetzer can you please also remove the outdated TODO note? (as commented above).

Then we can merge!

**Comment by 3nids** [comment]
argument names are part of the API since we have named arguments.
I am not the strictest one regarding API changes but this choice should be made consciously and be written in API changes dox.

**Comment by nyalldawson** [comment]
@3nids in this case it doesn't change anything, because these arguments are out arguments for python

**Comment by 3nids** [comment]
sorry, great then!

**Comment by tschmetzer** [comment]
> @tschmetzer can you please also remove the outdated TODO note? (as commented above).
> 
> Then we can merge!

Sorry for the delay! I was pretty busy and thought I had already removed it. Hope we're ready for a merge now!

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Thanks @tschmetzer , much appreciated!

**Pull request merged**